crates/core/src/config.rs

```diff
@@ -26,11 +27,33 @@ pub struct Config {
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct ServerConfig {
-    #[serde(default = "default_listen_addr")]
-    pub listen_addr: String,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub advertise_addr: Option<String>,
+#[serde(deny_unknown_fields)]
+#[serde(tag = "mode")]
+pub enum ServerConfig {
+    #[serde(rename = "standalone")]
+    Standalone {
+        #[serde(default = "default_dir")]
+        dir: PathBuf,
+        #[serde(default = "default_listen_addr")]
+        listen_addr: String,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        advertise_addr: Option<String>,
+    },
+    #[serde(rename = "cluster")]
+    Cluster {
+        #[serde(default = "default_dir")]
+        dir: PathBuf,
+        #[serde(default = "default_listen_addr")]
+        listen_addr: String,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        advertise_addr: Option<String>,
+        #[serde(default = "default_listen_peer_addr")]
+        listen_peer_addr: String,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        advertise_peer_addr: Option<String>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        initial_peer_addrs: Option<Vec<String>>,
```

Comment:
```suggestion
        initial_advertise_peer_addrs: Option<Vec<String>>,
```

Comes from https://etcd.io/docs/v3.3/op-guide/configuration/#--initial-advertise-peer-urls to make it clear.

Reply:
They are different semantics, but the name is acceptable.